osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -1433,188 +1463,407 @@ namespace renderkit {
-        q_xyz_quat_type q_roomFromHead;
-        if (params.roomFromHeadReplace != nullptr) {
-            /// Use the params.m_headFromRoom as our transform
-            q_from_OSVR(q_roomFromHead, *params.roomFromHeadReplace);
-        } else {
-            /// Use the state interface to read the most-recent
-            /// location of the head.  It will have been updated
-            /// by the most-recent call to update() on the context.
-            /// DO NOT update the client here, so that we're using the
-            /// same state for all eyes.
-            OSVR_TimeValue timestamp;
-            if (!m_headPoseCache || !m_headPoseCache->getLastReport(timestamp, m_roomFromHead)) {
-                // This it not an error -- they may have put in an invalid
-                // state name for the head; we just ignore that case.
-            }
-
-            // Do prediction of where this eye will be when it is presented
-            // if client-side prediction is enabled.
-            if (m_params.m_clientPredictionEnabled) {
-              // Get information about how long we have until the next present.
-              // If we can't get timing info, we just set its offset to 0.
-              float msUntilPresent = 0;
-              OSVR_RenderTimingInfo timing;
-              if (GetTimingInfo(whichEye, timing)) {
-                msUntilPresent +=
-                  (timing.timeUntilNextPresentRequired.seconds * 1e3f) +
-                  (timing.timeUntilNextPresentRequired.microseconds / 1e3f);
-              }
-
-              // Find out how long ago this tracker info was found.
-              float msSinceTrackerReport = 0;
-              {
-                  OSVR_TimeValue now;
-                  osvrTimeValueGetNow(&now);
-                  msSinceTrackerReport = static_cast<float>(osvrTimeValueDurationSeconds(&now, &timestamp) * 1e3);
-              }
-
-              // The delay before rendering for each
-              // eye will be different because they are at different delays past
-              // the next vsync.  The static delay common to both eyes has
-              // already been added into their offset.
-              float predictionIntervalms = msSinceTrackerReport +
-                msUntilPresent;
-              if (whichEye < m_params.m_eyeDelaysMS.size()) {
-                predictionIntervalms += m_params.m_eyeDelaysMS[whichEye];
-              }
-              float predictionIntervalSec = predictionIntervalms / 1e3f;
-
-              // Find out the pose velocity information, if available.
-              // Set the valid flags to false so that if to call to get
-              // velocity fails, we will not try and use the info.
-              OSVR_VelocityState vel;
-              vel.linearVelocityValid = false;
-              vel.angularVelocityValid = false;
-              if (osvrGetVelocityState(m_roomFromHeadInterface, &timestamp, &vel) != OSVR_RETURN_SUCCESS) {
```

Comment:
Again, get the velocity state for the eye here rather than the head in case we're using eyes and drop it in place of the head-space velocity.  And do an inner if() to see which eye, or else make a pointer and assign it to the current one before proceeding below to do the same code for both eyes.

Reply:
I think a pointer that points at head followed by a pair of if(eyes) { if (left) {repoint to left eye} else {repoint to right eye} } before a single osvrGetVelocityState() would be the clearest.

Reply:
addressed this a new commit